### ISSUE TITLE
Set Visual Studio solution filename

### ISSUE THREAD

**Issue by vinniefalco** [boostorg/boost#735]
If we could change CMakeLists.txt for the superproject then we can set the name of the solution otherwise it is always Boost.sln which is not ideal:
```
if( NOT DEFINED BOOST_PROJECT_NAME )
  set( BOOST_PROJECT_NAME "Boost" )
endif()
project( ${BOOST_PROJECT_NAME} VERSION 1.82.0 LANGUAGES CXX )
```

**Comment by vinniefalco** [comment]
The problem is that if you build more than one solution for different libraries, the Task Manager shows them all with the name Boost and you can't tell one from the other when switching

**Comment by pdimov** [comment]
No, sorry. The project name is significant and changing it just to rename the .sln file is not the right fix. You should ask in [the CMake discourse](https://discourse.cmake.org/) whether there's a way to change the name of the generated .sln file without changing the CMake project name.

**Comment by vinniefalco** [comment]
Yeah, I'll do that, but note that you are currently calling the resulting project "Boost" no matter what is in it (for example, even if you told cmake to only build one library). Although I guess that's one way of making "Boost" smaller.

**Comment by vinniefalco** [comment]
https://discourse.cmake.org/t/canonical-way-to-set-the-visual-studio-solution-sln-filename/7482

**Comment by pdimov** [comment]
"Boost" is the name of the superproject. The libraries have their own project names ("boost_lib").

**Comment by vinniefalco** [comment]
> "Boost" is the name of the superproject

Yes well my change renames the superproject...
